beater/beater.go

```diff
@@ -72,3 +70,27 @@ func (bt *beater) Stop() {
 	logp.Info("stopping apm-server...")
 	stop(bt.server, bt.config.ShutdownTimeout)
 }
+
+
+func validateConfig(beaterConfig *Config, beatConfig *beat.BeatConfig) error {
+	if beaterConfig.Frontend.isEnabled() {
+		if _, err := regexp.Compile(beaterConfig.Frontend.LibraryPattern); err != nil {
+			return fmt.Errorf("invalid regex for `library_pattern`: %v", err)
+		}
+		if _, err := regexp.Compile(beaterConfig.Frontend.ExcludeFromGrouping); err != nil {
+			return fmt.Errorf("invalid regex for `exclude_from_grouping`: %v", err)
+		}
+	}
+	if beatConfig != nil && beatConfig.Output.Name() == "elasticsearch" {
+		clients, err := es.NewElasticsearchClients(beatConfig.Output.Config())
+		if err != nil {
+			return fmt.Errorf("elasticsearch not available: %v", err)
+		}
+		for _, client := range clients {
+			if err := client.Connect(); err != nil {
+				return fmt.Errorf("elasticsearch not available: %v", err)
```

Comment:
This would mean if multiple ES connections are configured, but one is not available the server doesn't start. What benefit do you see from this?

Reply:
Yeah, maybe 1 up is enough as you suggest. I'm fine with that.

I did this small refactor also as a initial work on bootstrap checks, there are many more things we can add here (dependencies / incompatibilities between options, ssl, etc) but i don't have the full picture of how everything plays together.